src/CommonGLPI.php

```diff
@@ -1057,6 +1057,28 @@ class='btn btn-sm btn-icon btn-ghost-secondary'
    function display($options = []) {
       global $CFG_GLPI;
 
+      $redirect = false;
+
+      // Manage profile change
+      if (isset($_GET["change_profile"])) {
```

Comment:
No need to redirect if profile is same as the active one.
```suggestion
      if (isset($_GET["change_profile"]) && ($_SESSION['glpiactiveprofile']['id'] ?? -1) != $_GET["change_profile"]) {
```